src/sparseml/transformers/sparsification/trainer.py

```diff
@@ -119,6 +119,10 @@ def __init__(
         self.callback_disable_fp16 = DisableHalfPrecisionCallback(self)
         self.callback_handler.add_callback(self.callback_disable_fp16)
 
+        self.grad_sampler = GradSampler(
```

Comment:
thinking about some changes we can make to the grad sampler to make the requirements for the dataloader a little less strict. let's sync on this tomorrow. similar to the changes requested in the dataloader here: https://github.com/neuralmagic/sparseml/pull/541